### PR TITLE
enhance: Avoid use concrete segment type in segments interfaces

### PR DESCRIPTION
**Pull request by congqixia**
See also #34519

### CODE REVIEW

File: internal/querynodev2/segments/segment_loader.go [hunk 1]
```diff
@@ -1660,7 +1664,16 @@ func (loader *segmentLoader) getFieldType(collectionID, fieldID int64) (schemapb
 	return 0, merr.WrapErrFieldNotFound(fieldID)
 }
 
-func (loader *segmentLoader) LoadIndex(ctx context.Context, segment *LocalSegment, loadInfo *querypb.SegmentLoadInfo, version int64) error {
+func (loader *segmentLoader) LoadIndex(ctx context.Context,
+	// segment *LocalSegment,
```
Comment (XuanYang-cn) [inline internal/querynodev2/segments/segment_loader.go:1668]: please remove the commented code

File: internal/querynodev2/segments/segment_loader.go [hunk 2]
```diff
@@ -1087,9 +1086,14 @@ func (loader *segmentLoader) loadSealedSegment(ctx context.Context, loadInfo *qu
 }
 
 func (loader *segmentLoader) LoadSegment(ctx context.Context,
-	segment *LocalSegment,
+	// segment *LocalSegment,
```
Comment (XuanYang-cn) [inline internal/querynodev2/segments/segment_loader.go:1089]: ditto

File: internal/querynodev2/segments/segment_loader.go (hunk 1 above)
Comment (congqixia) [reply internal/querynodev2/segments/segment_loader.go]: done

File: internal/querynodev2/segments/segment_loader.go (hunk 2 above)
Comment (congqixia) [reply internal/querynodev2/segments/segment_loader.go]: done

**Pull request merged**